src/pt/argoscomics/src/eu/kanade/tachiyomi/extension/pt/argoscomics/ArgosComics.kt

```diff
@@ -1,63 +1,250 @@
+            .execute().body.string()
+        return regex.find(script)?.groupValues?.last(String::isNotBlank)
+            ?: throw IOException("Não foi possível encontrar token")
+    }
+
+    private fun getJsonBody(response: Response): String {
+        return MANGA_DATA_REGEX.find(response.body.string())?.groupValues?.last(String::isNotBlank)
+            ?: throw IOException("Não foi possível encontrar a lista de mangás")
+    }
+
+    private fun getScriptUrlFrom(url: String, cssSelector: String): String? {
+        val document = client.newCall(GET(url)).execute().asJsoup()
+        return document.selectFirst(cssSelector)?.absUrl("src")
+    }
+
+    companion object {
+        private val TEXT_PLAIN_MEDIA_TYPE = "text/plain;charset=UTF-8".toMediaTypeOrNull()
+        private val CHAPTERS_TOKEN_REGEX = """(?:unfavorite"[^\.]+.createServerReference\)\(")([^"]+)"(?:,.+getAllChapters)""".toRegex()
+
+        /**
+         * Combines multiple individual regular expressions to capture the "Next-Action"
+         * needed to call various API endpoints (popular, latest, search, details, pages).
+         *
+         * The token is a unique identifier (a string) found inside the 'createServerReference'
+         * function call within the page's JavaScript source, immediately preceding the
+         * corresponding backend function name (e.g., 'getAllWithoutFilters' for popular).
+         */
+        private val NEXT_ACTION_REGEX = buildList {
+            add("""(?:createServerReference\)\(")([^"]+)"(?:,.+getAllWithoutFilters)""".toRegex()) // popular token
+            add("""(?:"[^\.]+.createServerReference\)\(")([^"]+)"(?:,.+getLastUpdates)""".toRegex()) // latest token
+            add("""(?:(?:,.=.{2}\d+.){4}[^\.]+.createServerReference\)\(")([^"]+)"(?:,.+search)""".toRegex()) // search token
+            add("""(?:getAllChapters"[^\.]+.createServerReference\)\(")([^"]+)"(?:,.+getOne)""".toRegex()) // details token
+            add("""(?:"[^\.]+.createServerReference\)\(")([^"]+)"(?:,.+getPages)""".toRegex()) // pages token
+        }.joinToString("|").toRegex()
+
+        /**
+         * Combines regular expressions to identify and extract the **JSON string** containing
+         * the manga data (list, details, chapters, pages) from an HTTP response body.
+         *
+         * The purpose is to find the beginning of the valid JSON data, ignoring any
+         * preceding code or text. The pattern is defined by the expected first key-field
+         * of the JSON for each response type:
+         *
+         * - 'id': Starts the list of latest projects or search results (e.g., [{"id"...).
+         * - 'projects', 'title', 'groupName', 'pages': Starts other responses (e.g., {"projects"...).
+         */
+        private val MANGA_DATA_REGEX = buildList {
+            add("""\[\{"id".+""".toRegex())
+            add("""\{"(?:${listOf("projects", "title", "groupName", "pages").joinToString("|")})".+""".toRegex())
```

Comment:
It's easier to read if you don't mix DSLs, IMO.
```suggestion
            add("""\{"(?:projects|title|groupName|pages)".+""".toRegex())
```